hudi-common/src/test/java/org/apache/hudi/common/testutils/HoodieAdaptablePayloadDataGenerator.java

```diff
@@ -136,6 +148,18 @@ public static List<HoodieRecord> getDeletes(List<HoodieRecord> baseRecords, long
     return deletes;
   }
 
+  public static List<HoodieRecord> getDeletes(List<HoodieRecord> baseRecords, String newPartition, long ts, Class<?> payloadClass) throws IOException {
+    RecordGen recordGen = new RecordGen(payloadClass);
```

Comment:
do we have tests coverage for HoodieEmptyRecordPayload? 
thats the flow that spark-datasource 'deletes" take. so, lets ensure we have tests covering this as well.

Reply:
supported in https://github.com/apache/hudi/pull/8736/commits/e97c496d5636afb9ec4605e7232046f8e5edfb25